nifi-h2/nifi-h2-database-migrator/src/main/java/org/apache/nifi/dbmigration/h2/H2DatabaseUpdater.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.dbmigration.h2;
+
+import org.h2.jdbc.JdbcSQLNonTransientException;
+import org.h2.jdbcx.JdbcDataSource;
+import org.h2.mvstore.MVStoreException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+public class H2DatabaseUpdater {
+
+    private static final Logger logger = LoggerFactory.getLogger(H2DatabaseUpdater.class);
+
+    public static final String EXPORT_FILE_PREFIX = "export_";
+    public static final String EXPORT_FILE_POSTFIX = ".sql";
+    public static final String H2_URL_PREFIX = "jdbc:h2:";
+
+    private static final JdbcDataSource migrationDataSource = new JdbcDataSource();
```

Comment:
Although the `H2DatabaseUpdater` class does not appear to be used in a multi-threaded context, moving this declaration inside the `checkAndPerformMigration` method would avoid the potential for unexpected behavior with different connection parameters.